### ISSUE TITLE
Dialogs

### ISSUE THREAD

**Issue by guidezpl** [flutter/flutter#102811]

**Comment by mono0926** [comment]
A package for the same purpose has already been released and is so popular now

- [adaptive_dialog | Flutter Package](https://pub.dev/packages/adaptive_dialog)
- Demo: https://mono0926.com/adaptive_dialog

**Comment by rydmike** [comment]
Any day I can add `.adaptive` to yet another Flutter SDK widget is good day. Despite existing packages and my own internal version of this. I am for this and think it is a very desirable feature and hope to see it implemented 💙

**Comment by vbuberen** [comment]
> A package for the same purpose has already been released and is so popular now
> 
> - [adaptive_dialog | Flutter Package](https://pub.dev/packages/adaptive_dialog)
> - Demo: https://mono0926.com/adaptive_dialog


Yes, this is what is currently used by lots of people, but having an option to call .adaptive() (for example, like we can for CirculatProgressIndicator) is much better and straightforward.
